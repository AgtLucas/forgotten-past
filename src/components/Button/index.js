import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles.module.css';

const Button = ({ variant = 'default', children, ...rest }) => (
  <button
    className={styles[variant]}
    {...rest}
  >
    {children}
  </button>
);

Button.propTypes = {
  variant: PropTypes.oneOf(['default', 'emoji']),
  children: PropTypes.node.isRequired,
};

export default Button;