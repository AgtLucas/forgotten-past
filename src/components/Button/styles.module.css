.btn {
  width: 100%;
  height: 50px;
  border: 0;
  padding: 10px;
  border-radius: var(--borderRadius);
  cursor: pointer;
}

.btn:disabled {
  opacity: 0.6;
  cursor: default;
}

.default {
  composes: btn;
  background-color: var(--blue);
  color: var(--white);
}