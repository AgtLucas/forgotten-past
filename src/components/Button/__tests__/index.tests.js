import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Button from '../';

describe('Button', () => {
  it('should render Button', () => {
    const wrapper = shallow(<Button>Hey</Button>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should render an emoji type button', () => {
    const wrapper = shallow(
      <Button variant='emoji'>
        <span role='img' aria-label='My emoji button'>ðŸ¤˜</span>
      </Button>
    );

    expect(toJSON(wrapper)).toMatchSnapshot();
  })
});