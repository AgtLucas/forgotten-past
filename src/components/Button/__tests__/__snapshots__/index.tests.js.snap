// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render Button 1`] = `
<button
  className="default"
>
  Hey
</button>
`;

exports[`Button should render an emoji type button 1`] = `
<button
  className="emoji"
>
  <span
    aria-label="My emoji button"
    role="img"
  >
    ðŸ¤˜
  </span>
</button>
`;
